### PR TITLE
Beef up ring homomorphism to an algebra homomorphism task

### LINKED ISSUE #205: beef up a ring homomorphism to an algebra homomorphism
The completion machinery works for ring homomorphisms but not for algebra homomorphisms (as far as I know). We use it to complete the ring homomorphism K -> L to a ring homomorphism K_v -> L_w. But we need that this is a K-algebra homomorphism. The sorry in `adicCompletion_comap_algHom` in the file `FLT.DedekindDomain.FiniteAdeleRing.BaseChange` is data, but the data is all created in `adicCompletion_comap_ringHom`, and the claim about it being an algebra hom is just a claim about how the map commutes with the K-action.

### PR DESCRIPTION
**Pull request by WilliamCoram**
Claimed task from project

Closes #205

### CODE REVIEW

File: FLT/DedekindDomain/FiniteAdeleRing/BaseChange.lean [hunk 1]
```diff
@@ -169,13 +169,28 @@ noncomputable local instance (w : HeightOneSpectrum B) :
...
 variable {B L} in
 noncomputable def adicCompletion_comap_algHom (w : HeightOneSpectrum B) :
-    letI : Algebra K (adicCompletion L w) := RingHom.toAlgebra <|
-  (algebraMap L (adicCompletion L w)).comp (algebraMap K L);
-    letI : Module K (adicCompletion L w) := Algebra.toModule
-    (HeightOneSpectrum.adicCompletion K (comap A w)) →ₐ[K] (HeightOneSpectrum.adicCompletion L w) :=
-  sorry -- use `adicCompletion_comap_ringHom` and prove it's a K-algebra homomorphism
+    (HeightOneSpectrum.adicCompletion K (comap A w)) →ₐ[K] (HeightOneSpectrum.adicCompletion L w) := by
+    use adicCompletion_comap_ringHom A K w
+    intro r
```
Comment (kbuzzard) [inline FLT/DedekindDomain/FiniteAdeleRing/BaseChange.lean:177]:
```suggestion
    (HeightOneSpectrum.adicCompletion K (comap A w)) →ₐ[K]
    (HeightOneSpectrum.adicCompletion L w) where
  __ := adicCompletion_comap_ringHom A K w
  commutes' r := by
```
What you write compiles, but this is the idiomatic syntax. I also break the line which was over 100 characters.

File: FLT/DedekindDomain/FiniteAdeleRing/BaseChange.lean [hunk 2]
```diff
@@ -169,13 +169,28 @@ noncomputable local instance (w : HeightOneSpectrum B) :
...
+    simp only [RingHom.toMonoidHom_eq_coe, OneHom.toFun_eq_coe, MonoidHom.toOneHom_coe,
+      MonoidHom.coe_coe]
+    have : (adicCompletion_comap_ringHom A K w) (r : adicCompletion K (comap A w))  =
+    (algebraMap L (adicCompletion L w)) (algebraMap K L r) := by
```
Comment (kbuzzard) [inline FLT/DedekindDomain/FiniteAdeleRing/BaseChange.lean:181]:
```suggestion
        (algebraMap L (adicCompletion L w)) (algebraMap K L r) := by
```

File: FLT/DedekindDomain/FiniteAdeleRing/BaseChange.lean [hunk 3]
```diff
@@ -169,13 +169,28 @@ noncomputable local instance (w : HeightOneSpectrum B) :
...
+      letI : UniformSpace L := w.adicValued.toUniformSpace;
+      letI : UniformSpace K := (comap A w).adicValued.toUniformSpace;
```
Comment (kbuzzard) [inline FLT/DedekindDomain/FiniteAdeleRing/BaseChange.lean:183]:
```suggestion
      letI : UniformSpace L := w.adicValued.toUniformSpace
      letI : UniformSpace K := (comap A w).adicValued.toUniformSpace
```
Semicolons aren't needed.

File: FLT/DedekindDomain/FiniteAdeleRing/BaseChange.lean [hunk 4]
```diff
@@ -169,13 +169,28 @@ noncomputable local instance (w : HeightOneSpectrum B) :
...
+      rw [adicCompletion_comap_ringHom, UniformSpace.Completion.mapRingHom]
+      have h : @UniformSpace.Completion.coe' K this r  = (r : adicCompletion K (comap A w)) := rfl
+      rw [← h, UniformSpace.Completion.extensionHom_coe
+      (UniformSpace.Completion.coeRingHom.comp (algebraMap K L))
+      (UniformSpace.Completion.mapRingHom.proof_2 (algebraMap K L)
+      (adicCompletion_comap_ringHom.proof_4 A K w) :
+      Continuous ⇑(UniformSpace.Completion.coeRingHom.comp Algebra.toRingHom)) r]
+      rfl
```
Comment (kbuzzard) [inline FLT/DedekindDomain/FiniteAdeleRing/BaseChange.lean:191]:
```suggestion
      rw [show (r : adicCompletion K (comap A w)) = @UniformSpace.Completion.coe' K this r from rfl]
      apply UniformSpace.Completion.extensionHom_coe
```
All this `proof_2` and `proof_4` stuff is somehow known to the elaborator (to be honest I don't quite know what's going on here, but this proof works).

**Review by kbuzzard — APPROVED** [review]
Thanks! I made some suggestions which make the code more idiomatic but if you're happy with the suggestions then feel free to commit them and then I'll merge.

**Comment by WilliamCoram** [comment]
Cheers! I have committed your suggestions.

**Pull request merged**